### PR TITLE
refactor: adding tests, updating grappa.getProductId

### PR DESCRIPTION
**Pull request by dsshap**
Related to:

https://github.com/antoncoding/grappa/issues/126

### CODE REVIEW

File: src/core/Grappa.sol [hunk 1]
```diff
@@ -118,6 +118,27 @@
...
+            uint64 expiry,
+            uint64 longStrike,
+            uint64 shortStrike
+        )
+    {
+        (TokenType _tokenType, uint40 _productId, uint64 _expiry, uint64 _longStrike, uint64 _shortStrike) = TokenIdUtil
+            .parseTokenId(_tokenId);
+
+        return (_tokenType, _productId, _expiry, _longStrike, _shortStrike);
+    }
```
[the remark this answers is not in the document]
Comment (antoncoding) [reply src/core/Grappa.sol:140]: @dsshap can you make it external?

[the remark this answers is not in the document]
Comment (antoncoding) [reply src/core/Grappa.sol:140]: seems like the warning is correct, it can be marked as external

File: src/test/unit-tests/GrappaRegistry.t.sol [hunk 2]
```diff
@@ -65,6 +67,27 @@ contract GrappaRegistry is Test {
...
         assertEq(collatDecimals, 18);
     }
+
+    function testReturnOptionDetailsFromTokenId() public {
+        uint256 _expiry = block.timestamp + 14 days;
+        uint256 _strikePrice = 4000 * UNIT;
+
+        grappa.registerAsset(address(weth));
+
+        uint40 _product = grappa.getProductId(address(0), address(0), address(weth), address(0), address(weth));
```
Comment (antoncoding) [inline src/test/unit-tests/GrappaRegistry.t.sol:77]: I'm currently only prefixing function arguments with `_`. Maybe better to keep things consistent.

**Review by antoncoding — APPROVED** [review]
LGTM, this format makes better sense

File: src/core/Grappa.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (dsshap) [reply src/core/Grappa.sol:140]: @antoncoding it cant be marked as external because it is being referenced on L359.

**Comment by antoncoding** [comment]
@dsshap feel free to merge

**Pull request merged**